### ISSUE TITLE
stability: panic while rebalancing away from node.

### ISSUE THREAD

**Issue by mberhault** [cockroachdb/cockroach#18777]
sha: fbcd8ca7350a79e91f635fe59f8c5f74bde79eba

While playing with zone configs on `indigo`, `cockroach-indigo-0012` encountered a panic while getting ranges rebalanced off of it:
```
I170926 13:37:07.621695 83657629 sql/mon/bytes_usage.go:578  [n9] admin: bytes usage increases to 130 KiB (+10240)
I170926 13:37:07.662885 83657629 sql/mon/bytes_usage.go:578  [n9] admin: bytes usage increases to 260 KiB (+10240)
panic: session: unexpected 116650 leftover bytes

goroutine 83656411 [running]:
github.com/cockroachdb/cockroach/pkg/sql/mon.(*BytesMonitor).doStop(0xc435c27d38, 0x2b1d0a0, 0xc4308c34c0, 0xc428a47101)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/mon/bytes_usage.go:388 +0x3af
github.com/cockroachdb/cockroach/pkg/sql/mon.(*BytesMonitor).Stop(0xc435c27d38, 0x2b1d0a0, 0xc4308c34c0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/mon/bytes_usage.go:375 +0x44
github.com/cockroachdb/cockroach/pkg/sql.(*Session).Finish(0xc435c27000, 0xc420714480)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/session.go:551 +0x1cc
github.com/cockroachdb/cockroach/pkg/server.(*adminServer).RangeLog(0xc420232500, 0x2b1d160, 0xc42fda3b60, 0xc4329406a0, 0xc427b719e0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/server/admin.go:837 +0x1827
github.com/cockroachdb/cockroach/pkg/server/serverpb._Admin_RangeLog_Handler.func1(0x2b1d160, 0xc42fda3aa0, 0x1bb5c00, 0xc4329406a0, 0x1bb5c00, 0xc4329406a0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/server/serverpb/admin.pb.go:1589 +0x89
github.com/cockroachdb/cockroach/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1(0x2b1d160, 0xc42fda3aa0, 0x1bb5c00, 0xc4329406a0, 0xc4311f2340, 0xc4311f2360, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc/server.go:48 +0xa54
github.com/cockroachdb/cockroach/pkg/server/serverpb._Admin_RangeLog_Handler(0x1c84200, 0xc420232500, 0x2b1d160, 0xc42fda3aa0, 0xc4314e4cd0, 0xc420295740, 0x0, 0x0, 0x5bfab7, 0xc4313875c0)
        /go/src/github.com/cockroachdb/cockroach/pkg/server/serverpb/admin.pb.go:1591 +0x16d
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc420094000, 0x2b26fa0, 0xc420da2000, 0xc42a025900, 0xc42080ec90, 0x2aea480, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:781 +0xab4
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleStream(0xc420094000, 0x2b26fa0, 0xc420da2000, 0xc42a025900, 0x0)
        /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:981 +0x1528
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc4202277a0, 0xc420094000, 0x2b26fa0, 0xc420da2000, 0xc42a025900)
        /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:551 +0x9f
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
        /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:549 +0xa1

```

`cockroach-indigo-001[0-2]` have the `locality=northeurope` flag set.
at 2017-09-26-13:29UTC, I set the `[-locality=northeurope]` on the `test` database (used by the kv load generator).
ranges were getting rebalanced away from the three nodes in that region (including `cockroach-indigo-0012` until it crashed at 13:37:
![draining-indigo](https://user-images.githubusercontent.com/5901480/30863916-3d87fd16-a2a0-11e7-86a1-593946a76566.png)

**Comment by mberhault** [comment]
@knz: this looks like a session accounting issue. Could you take a look?

**Comment by mberhault** [comment]
Full stderr: [cockroach.stderr.gz](https://github.com/cockroachdb/cockroach/files/1333416/cockroach.stderr.gz)

**Comment by knz** [comment]
This was addressed by #22685.
